plugin/pkg/admission/priority/admission_test.go

```diff
@@ -414,3 +414,175 @@ func TestPodAdmission(t *testing.T) {
 		}
 	}
 }
+
+func TestUpdatePodAdmission(t *testing.T) {
```

Comment:
I think this test is overly complex for testing update and does not test "Update". All that update does is to leave integer value of priority intact if PodSpec.PriorityClassName has not changed.
So, the test scenario should be like this:
1. Create a PriorityClass.
2. Create a pod with PriorityClassName set to the name of the priority class you created in step 1.
3. Delete the PriorityClass and add another one with the same name, but a different value.
4. Update the pod without changing its PriorityClassNaem.
5. Make sure that the integer value of Priority in PodSpec has not changed.

Reply:
Done.